src/lib.rs

```diff
@@ -1880,7 +1866,7 @@ impl<
     pub async fn balance(&self, asset: &AssetCode) -> U256 {
         let KeystoreSharedState { state, model, .. } = &*self.read().await;
         state.balance(
-            model.sending_accounts.pub_keys(),
+            model.sending_accounts.iter_pub_keys().collect(),
```

Comment:
Could maybe change `state.balance` to take `impl IntoIterator<Item = UserAddress>` instead of `Vec<UserAddress>` if all it does is iterate over the addresses